### ISSUE TITLE
Out of Sync issue when using embeded subtitles

### ISSUE THREAD

**Issue by n00ble** [UniversalMediaServer/UniversalMediaServer#1123]
Thanks for great software

This may be an old issue. I have basic skills so I was not able to check of duplicate reporting. I tried to find a solution to no avail. I have uploaded the shortest file that does this too.

My wife recently bought an old series "3rd Rock From The Sun". I ripped the CDs so to be able to view them easier on the TV. I used Handbrake 1.0.1 and included the subtitles. I have tried to change the ripping variables to no avail. I have also tried to disable "Keep AC-3 tracks" in "Audio Settings" under "Transcoding Settings", but the out of sync still occurs.

The movie starts with the fast forward and backward options greyed out. Few seconds later it gets like a hiccup and things go out of sync. I tried to run UMS with Subtitles disabled, and the voice is in-sync but it exits the episode like 5 minutes later...

I am using UMS on a WiFi network with a Samsung TV (UA50JU6400).
I have uploaded the shortest file: "UMS Bug Out of Sync 3rd Rock S6d1-15.mp4"

"UMS Bug Out of Sync 3rd Rock S6d1-15.7z"
MD5: f8a738c89e726a28fbcceb87924aedc7
http://ge.tt/7arYU9i2

Please let me know if there is a way to avoid these issues.

Thnx

**Comment by n00ble** [comment]
The same bug is still present in version 6.5.3

**Comment by n00ble** [comment]
The same issue is still present in version 6.6.0

**Comment by Nadahar** [comment]
@n00ble I'm not expert at transcoding issues, but it seems you have to little memory:
```powershell
Muxer frame buffer cannot allocate memory!
```

2.2 GB on Windows 10 is extremely low, you should upgrade to atleast 4 GB or downgrade to a less memory-hungry OS.

You can also try to turn of the "Defer to MEncoder" option for FFmpeg in the transcoding settings so that FFmpeg will be used instead, but I'm not sure anything can work with this little memory.

**Comment by Sami32** [comment]
@n00ble I agree on that:
Deselect the "Defer to MEncoder when transcoding problematic subtitles" option found in the "Transcoding settings" tab in "FFmpeg" video files engines section.

Avoid MEncoder, as it an old encoder far less efficient that FFmpeg, and near to not maintened anymore. So i'll suggest you to use only for CD/DVD discs.
And as you see it give you unecessary errors as well.

I recommend you to force the renderer called "Samsung JU6400", that will fit your renderer much better.

**P.S.** I'm curious to know if UMS (Win10 & 8GB) work fine with your _Bose SoundTouch 30_ renderer ?

**Comment by n00ble** [comment]
Thank you for the reply, and I appreciate your efforts and feedback. I am not always at home and testing these stuff takes time :\

The system has 32 GB of RAM. I think that thr 2.2 GB is the allocated memory for UMS (ie. double the default of 1280 MB). The issue persists with a UMS memory of 6400 MB (5 times the dafault setting).

The skipping (de-sync) occurs within 2 sec of starting the clip. i.e. may not be memory related since the clip itself is like 7 MB only too. I have not had these issues with full movies from Torrents...

I used HandBrake to rip the DVDs, and the subtitles were integrated into the MP4 by default (not burned or hardcoded). I have yet to figure out a way to extract the Subtitles into a separate .srt file. I have not edited any of the videos. Ripping 17 DVDs took a long time and was not fun. I am not enthusiastic to do it all over again. I still have a couple of other DVD sets to rip, but I will make an effort to figure out how to avoid these issues first; may be by finding a way to have the subtitles in a separate file.
I thought that the debug file would have the PC specs in it. The computer specs summary from "Speccy" software are below. For me it is on the higher end of PC specs:
Summary:
		Operating System
			Windows 10 Pro 64-bit
		CPU
			Intel Core i7 6700K @ 4.00GHz	20 °C
			Skylake 14nm Technology
		RAM
			32.0GB Dual-Channel Unknown @ 1069MHz (15-15-15-35)
		Motherboard
			MSI Z170A GAMING M5 (MS-7977) (U3E1)	30 °C
		Graphics
			HP ENVY 32 (2560x1440@59Hz)
			NVIDIA GeForce GTX 960 (ASUStek Computer Inc)	32 °C
		Storage
			223GB KINGSTON SV300S37A240G (SSD)	24 °C
			1863GB Western Digital WDC WD2003FZEX-00Z4SA0 (SATA)	26 °C
		Optical Drives
			HL-DT-ST DVDRAM GH24NSD1
		Audio
			NVIDIA High Definition Audio
			
In Tab "Transcoding Settings" > FFmpeg
I unchecked the option "Defer to MEncoder when transcoding problematic subtitles".

I edited the "Samsung-JU6400.conf" file and made the adjustments mentioned in:
https://github.com/UniversalMediaServer/UniversalMediaServer/pull/1207/files

I tried to adjust the "renderer_force_default = "Samsung JU6400" in UMS.conf
I tried Samsung JU6400", "Samsung-JU6400",  "Samsung-JU6400.conf", and without "" 

The issue persists.

I have not used my Bose SoundTouch 30 speaker with UMS. It is on the same WiFi network. It would be nice to be able the FLAC files on it. I currently use it either for streaming online radios (which is a waste...) or to play FLAC files from an FiiO X3-II music player (via AUX). I plan to try to find a way to play FLAC music files via UMS or Kodi.
			
Thank you again

**Comment by Sami32** [comment]
@n00ble I need to have your new trace log with these new changes used.

After solved that issue we can check at your Bose if your want.

**Comment by Nadahar** [comment]
@n00ble Ok, my fault about the memory - I though it was the system memory, not the JVM memory displayed. Still, MEncoder complained that it couldn't allocate memory, and that would explain a lot. What's interesting is what FFmpeg will say now that you've turned of "Defer to MEncoder". So, as @Sami32 says, new logs plx 😉

**Comment by n00ble** [comment]
An additional finding (may be of no technical significance) is that shortly after the "voice de-sync" starts, and while the clip is still playing, the TV shows disconnection message.

I made the changes to the "Samsung-JU6400.conf" file by editing it.

I am not sure that the files are 4K. http://ge.tt/7arYU9i2  
I remember that the DVD resolution I saw in HandBrake was like 720p or something like that. I do not need a higher resolution for these series. I doubt the WiFi would support this resolution; that may require a direct Ethernet cable (PC to router and router to TV).

[ums_dbg_6.6.0_FFmpeg-Unchecked-Defer to MEncoder_Samsung JU6400 Edited.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/805024/ums_dbg_6.6.0_FFmpeg-Unchecked-Defer.to.MEncoder_Samsung.JU6400.Edited.zip)

I may not be able to do any additional changes/debugging on this machine for the coming ~ 2 wks.

Thanks again.

**Comment by Sami32** [comment]
@n00ble Did everything return to "normal" while disabling the subtitles ? (in "Transcoding settings" tab)
You can also test from your #--TRANSCODE--# folder, your video with/without subtitle.

Do you have ArialMT fonts in your Windows 10 ?

An other test could be to go into the "Transcoding settings" tab and deselect the "use embedded style" option in the "Subtitles settings" sub tab.
After that, select the "FFmpeg" video files engines on the left and select "Use font settings...".

P.S. As a remark for the next time, you should specify the audio/subtitle language when converting your videos files.
"Lossless DVD video playback" should be selected, if you want keep quality and CPU use saving.

**Comment by Sami32** [comment]
@Nadahar Did this ID come from a WEB.conf videostream ?
> DEBUG 2017-02-27 23:06:23.091 [New I/O worker #9] net.pms.dlna.GlobalIdRepo GlobalIdRepo: id not found: 76
> DEBUG 2017-02-27 23:06:23.091 [New I/O worker #9] net.pms.dlna.GlobalIdRepo GlobalIdRepo: id not found: 76
> DEBUG 2017-02-27 23:06:23.091 [New I/O worker #9] net.pms.dlna.DLNAResource Bad id 76 found in path

Do you think you can fix it ?
> INFO  2017-02-27 23:06:11.935 [cling-8] net.pms.util.SystemErrWrapper [Fatal Error] :1:1095: XML document structures must start and end within the same entity.
> DEBUG 2017-02-27 23:06:11.935 [cling-8] net.pms.network.UPNPControl Error parsing xml: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1095; XML document structures must start and end within the same entity.
> INFO  2017-02-27 23:06:11.941 [cling-10] net.pms.util.SystemErrWrapper [Fatal Error] :1:154: XML document structures must start and end within the same entity.
> DEBUG 2017-02-27 23:06:11.942 [cling-10] net.pms.network.UPNPControl Error parsing xml: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 154; XML document structures must start and end within the same entity.

**Comment by n00ble** [comment]
Hello again,

As I mentioned above, when I disabled the subtitles in the "Transcoding Settings" > "Common Transcoding Settings", the video played normally for few minutes, and then the program exited unexpectedly.

I think the regular Arial font in Windows is a MonoType font. To be sure, I searched online for, downloaded, and installed ArialMT font. Exited then restarted UMS. Nothing has changed.

I applied the mentioned changes to "Transcoding Settings" Tab:
- Unchecked "Use Embedded Style"
- FFmpeg Checked "Use Font Settings"
Exited then restarted UMS. Nothing has changed.

Irrespective of the coding adjustments, UMS should not result in these errors.
Let me know if you need me to test some more settings or if you need the log files of any specific adjustments. I can do some testing for the coming two days :\

Thnx again for your efforts and help

**Comment by Nadahar** [comment]
@Sami32 
> Do you think you can fix it ?

No, the error itself isn't a "problem", the way we handle it might be. I doubt it's related to this issue anyway. The error is an XML parsing error, and it has to do with the renderer remote control - which probably doesn't work properly for this renderer because of it. Without checking, the problem is most likely that the renderer sends some invalid XML. I really don't want to touch the player remote control stuff, as that's another mine-field.

**Comment by Sami32** [comment]
@n00ble Yes, please replace your ums.jar by that one, test it and report any changes while joining your full trace log:
https://we.tl/tzIqwp0pLj

About the xml parsing error, i saw that in some renderers from the same family as your, i will need a Wireshark session. So if you could post it, that should help me to go through that and see if i can fix it.
That's said it's not related to your issue, it's why i didn't address it to you.

**Comment by n00ble** [comment]
Hello,

I replaced the ums.jar with the downloaded one (after backup). Loss of sync still occurs. This was done without the changes in my previous post.
[ums_dbg_replaced_UMS-jar.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/832344/ums_dbg_replaced_UMS-jar.zip)

I will have to install Wireshark... I don's have that much experience or skills in that field. I can try to do that later today or early tomorrow.
Thnx

**Comment by Sami32** [comment]
Just to be sure, your out of sync issue happen to you only with your homemade convertion videos, right ?

About Wireshark easy steps, you can have a look there:
http://www.universalmediaserver.com/forum/viewtopic.php?f=9&t=9126&p=29350&hilit=Wireshark#p29350
http://www.universalmediaserver.com/forum/viewtopic.php?f=5&t=1563&p=29725&hilit=Wireshark#p29725

**Comment by Sami32** [comment]
OK, i guess i check your posts too late ;)
It seem that the renderer used was done badly, so replace the one you use with one of these 2 and give feedback, please:
1) [Samsung-JU6400.conf.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/832630/Samsung-JU6400.conf.txt)
2) [Samsung-JU6400.conf.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/832637/Samsung-JU6400.conf.txt)

As it was quickly modified from the original file for testing purpose only, don't expect accuraty or exhaustive formats support.

**Comment by n00ble** [comment]
Hello,

I tried both files; replacing the "Samsung-JU6400.conf".

1. With subtitles enabled; Same issue. 
With subtitles disabled, the clip playes notmally. The forward and backward buttons are not greyed out. The next episode is ~ 21 min played normally till 4 min 47 sec. The controls remain there as if it is still playing. Pressing pause/play (on the TV) does not change anything.

2. No subtitles showed (even with the subtitles enabled). The clip played without issues. The forward and backward buttons are not greyed out. The ~ 21 min episode played normally till it stopped ~ 6 min. The controls are still as if it is playing. Pressing pause/play (on the TV) does not change anything.
With subtitles disabled, the clip playes notmally. The forward and backward buttons are not greyed out. The next episode is ~ 21 min played normally till 4 min 51 sec. The controls remain there as if it is still playing. Pressing pause/play (on the TV) does not change anything.

With the original file and with subtitles disabled, the clip played fine, and the 21 min episode played till ~ 3 mins where it exited by itself.

I installed Wireshark, and I included the filter set.
The resulting file is:
UMS_Out-of-Sync_Wireshark.7z
http://ge.tt/8LylOCj2

Thnx again

**Comment by Sami32** [comment]
Thank you for the Wireshark capture, i'll check at it, and hopefully if i cannot handle it, an issue could be open.

So, if i understood your answer, you get this issue only with your own homemade videos; that doesn't sound good for you...

**Comment by n00ble** [comment]
The error occurs with the ones that I ripped. Its with the set of DVDs that I have for this series. 

It rarely occurs with downloaded movies, and I think that occurred only once in the past 6 months or so.

I will have to find a way to rip DVDs with separate .srt files using Handbrake.

Thnx

**Comment by Sami32** [comment]
I don't know if it come from Handbrake itself or/and users skills, but the Google relate too much users homemade videos issue realized with Handbrake.

So perhaps you should consider having a look at [MP4Box](https://www.videohelp.com/software/MP4Box) and [MKVToolnix](https://www.videohelp.com/software/MKVToolNix) ;)

**Comment by Nadahar** [comment]
Handbrake isn't supposed to require much "skill", that is the whole point. Whatever the problem is, its probably some combination of codec/codec versions and other settings that FFmpeg doesn't like. In such a "fluid" situation it's hard to point out what's "at fault" - the important thing is rather how to avoid it.

**Comment by Sami32** [comment]
@n00ble The last chance to get it work for you will be to try with the [latest FFmpeg version](https://ffmpeg.org/download.html). Download it, backup the UMS version used and put the new FFmepg executable there and try again.
Other way you will need to report this issue on the FFmpeg bug tracker board.

As a work around, if the modified sample that i post here work for you, you can have a chance to save you the effort of reencoding all your videos:
[UMS.mp4.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/835798/UMS.mp4.txt)

Hoping that help...Whatever, take time to give feedback, please.

**Comment by Sami32** [comment]
@Nadahar Handbrake is know for making MP4 subtitles not well supported/standardized, at opposite MP4Box do this very well.

**Comment by SubJunk** [comment]
@Slarper yes unfortunately FFmpeg still has the same limitation of parsing the entire file before it burns subtitles, so we have to use MEncoder by default. You can however manually use FFmpeg via the `#--TRANSCODE--#` folder. On some files, especially very large files, you might have to attempt playback more than once, because the first time FFmpeg will build its subtitles cache for the file and your media player will likely timeout while that happens. But you should have in-sync subtitles when it is finished
